ls1intum/Artemis
# Enhancement/guided tour/enable user interaction

### Checklist

### Description
This enhancement allows users to interact with elements in the highlighted section while the guided tour is active. This user interaction is e.g. necessary when the user has to click certain buttons.
If `enableUserInteraction` is enabled for a tour-step then the Next-Button should be disabled and the user can only navigate to the next step.
This implementation is required for the guided-tour for the code-editor and will be extended there.

New UI changes:
- The highlighted elements have a border to be more recognisable
- A hint text will be displayed if a click interaction is required
- A fade animation is added so that the transition between tour steps is smoother

### Steps for Testing
1. Log in to Artemis
2. Start tour for course overview page
3. Click anywhere on the screen outside the tour dialog
4. It should not trigger any actions
5. Navigate to step 6 `Personal settings`
6. Account menu is clickable and the user cannot navigate to the next step

### Screenshots
![video](https://user-images.githubusercontent.com/33764166/64780694-5b311c80-d561-11e9-8491-abcfc2e06d3a.gif)
